pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -116,12 +116,32 @@ def __init__(
 
         self._init_deterministic(deterministic)
 
+        if num_processes:
+            rank_zero_deprecation(
+                f"Setting `Trainer(num_processes={num_processes})` is deprecated in v1.6 and will be removed"
+                " in v1.8. Please use `Trainer(accelerator='cpu', devices={num_processes})` instead."
```

Comment:
From the issue:
> if there are no major quelms about this and everyone's excited, let's roll it out for 2.0

This won't be removed in v1.8

Reply:
Oh, my understanding from @tchaton's comment https://github.com/PyTorchLightning/pytorch-lightning/issues/10410#issuecomment-988644568 is that we will remove in v1.8.

> I don't believe this change requires a Lightning 2.0 as this is a natural evolution of Lightning becoming hardware-agnostic directly at the Trainer level.
> IMO, I would like to action this change for v1.6.

@tchaton can you confirm?

Reply:
As far as I know the consensus is deprecated in 1.6 and removed in 2.0

Reply:
I think it is safer to use 2.0 for now. We will see how this evolves.